department-of-veterans-affairs/caseflow
# Piedram/appeals 37878

Resolves:

# Description
As a Mail team user I need to fill out response letter information so that I can add the response letter to a correspondence. 
 
## Acceptance Criteria

When the "+add letters" button is clicked a response letter tile appears with the below sections:

1 Date Sent
  - This is defaulted to the current date
  - This has a date picker
  - Required
 
2 Letter Type: Dropdown
   - Is populated with values in attached document
   - Required

3 Letter Title 
   - This dropdown is disabled until a Letter Type is selected
   - Required

4 Letter sub-category 
  - This dropdown is disabled until a Letter Title is selected
  - Is populated with values in attached document
  - Required

5 Letter sub-category reason 
  - This dropdown is disabled until a sub-category is selected
  - Is populated with values in attached document
  - Required only if Letter sub-category has values


6 How long should the response window be for this response letter? 
   6.1  Radio button
   6.2 Custom value needs to be between 1-64
       - "Number of days (Value must be between 1 and 64)"
       - Text value
   6.3 Options are based on above selections in attached document
   6.4 Required

7 User can add additional response letters
  - User can add up to 3 response Letters

8 User can remove response letters

9 If a response letter button is clicked, "Continue" button is grayed out until all dropdowns are filled out or user removes response letter


## Testing Plan

1. Login as a Mail team user `MAIL_TEAM_SUPERVISOR_MAIL_INTAKE_USER (VACO)`
2. From the App selector, select and click on `Queue`
3. In Your cases page, click on `Switch views` and select `Correspondence Cases`
4. Now you have to see the `Correspondence Cases` page
5.  Click in one case, you can see the `Review Package` page
6. Scroll to the bottom of the page and click on `Create record` button
7. Now you can see the first tab on `intake` page and see the `+ Add Letter` button

![image (28)](https://github.com/department-of-veterans-affairs/caseflow/assets/110848569/6dc269ff-714c-49ef-b7f1-e3109d6b3d54)

8. When you click on `+ Add Letter` a new letter appears, and the continue button disappears until you complete the entire form.
9. You can Add only 3 letters for correspondence or less when you have 3 letters the `+ Add Letter` button is not clickable anymore.
10. Use the `Remove letter` link to remove the letter.

![image (29)](https://github.com/department-of-veterans-affairs/caseflow/assets/110848569/16469d7e-4af6-4c80-9006-6b8ea77d1a4a)

![image (30)](https://github.com/department-of-veterans-affairs/caseflow/assets/110848569/706f7b58-6c49-41c0-b34c-ea32d894e971)

11. The Windows response, depends on the `letter type` or `letter title` so when you click in this selector the response windows show up with the options

![image (31)](https://github.com/department-of-veterans-affairs/caseflow/assets/110848569/34e19601-779b-4792-9a7d-d2faf246cc3a)

![image (32)](https://github.com/department-of-veterans-affairs/caseflow/assets/110848569/443728b2-4765-40bd-a65c-099edfde51cc)

![image (33)](https://github.com/department-of-veterans-affairs/caseflow/assets/110848569/35099ad6-08eb-4fba-9ee1-b01f67a11e4b)

![image (34)](https://github.com/department-of-veterans-affairs/caseflow/assets/110848569/c9cf6822-679d-4778-adcd-95336d3fe4ee)

12. When the response window is `65 days` you have a custom option too.

![image (35)](https://github.com/department-of-veterans-affairs/caseflow/assets/110848569/521e6252-2243-49fa-b7d1-bf3995624cc4)


![image (36)](https://github.com/department-of-veterans-affairs/caseflow/assets/110848569/7e0b8d43-1dfb-4fdc-b251-2c53b1872e32)
